### PR TITLE
improved size of small asteroids

### PR DESCRIPTION
**Pull request by mgarraha**
### Description
Most asteroids' sizes are unknown and estimated from their absolute magnitude.
Previous versions would round a small asteroid up to a radius of 1 km and save that in ssystem_minor.ini.
This caused small asteroids' Info to show an incorrect "Diameter: 2 km."

MPC elements do not provide albedo or radius, so in this revision,
SolarSystemEditor::readMpcOneLineMinorPlanetElements no longer sets them.
SolarSystem::loadPlanets calls new method MinorPlanet::updateEquatorialRadius,
which estimates radius using the same formula as before but does not round up.
To fix a bad radius already in ssystem_minor.ini, delete that line manually.
If an asteroid's size is known and different from the usual estimate, a radius may be added manually.

Fixes # (issue)

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a documentation update
- [ ] Housekeeping

### How Has This Been Tested?
With Size enabled in Info:

* Newly imported asteroid with H>18 shows reasonable diameter, e.g. 2024 ON with H=20.5 shows 0.3 km
* Asteroid imported using older release shows reasonable diameter if radius is deleted from ssystem_minor.ini
* Asteroid with radius specified in ssystem_minor.ini shows 2x that diameter
* Newly imported comet still shows 10 km diameter, e.g. C/2024 S1

**Test Configuration**:
* Operating system: Linux Ubuntu 22.04
* Graphics Card: NVidia GeForce GT 750M, driver 390

## Checklist:
- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (header file)
- [ ] I have updated the respective chapter in the Stellarium User Guide
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/core/modules/MinorPlanet.hpp [hunk 1]
```diff
@@ -93,6 +93,9 @@ class MinorPlanet : public Planet
 	//! @param slope Slope parameter G. This is usually [0..1], sometimes slightly outside. Allowed here [-1..2].
 	void setAbsoluteMagnitudeAndSlope(const float magnitude, const float slope);
 
+	//! sets radius based on absolute magnitude.
```
Comment (gzotti) [inline src/core/modules/MinorPlanet.hpp:96]: ... based on absolute magnitude and albedo (assume default 0.15)

File: src/core/modules/SolarSystem.cpp [hunk 2]
```diff
@@ -1266,10 +1266,10 @@ bool SolarSystem::loadPlanets(const QString& filePath)
 			const QString horizonMapName = ( hidden ? "" : englishName.toLower().append("_normals.png")); // no normal maps for invisible objects!
 
 			newP = PlanetP(new MinorPlanet(englishName,
-						    pd.value(secname+"/radius", 1.0).toDouble()/AU,
+						    pd.value(secname+"/radius", 0.0).toDouble()/AU,
 						    pd.value(secname+"/oblateness", 0.0).toDouble(),
 						    color, // halo color
-						    pd.value(secname+"/albedo", 0.25f).toFloat(),
+						    pd.value(secname+"/albedo", 0.15f).toFloat(),
```
Comment (gzotti) [inline src/core/modules/SolarSystem.cpp:1272]: While this is in line with the estimate, is there a citable source for this estimate?

File: plugins/SolarSystemEditor/src/SolarSystemEditor.cpp [hunk 3]
```diff
@@ -1195,13 +1195,7 @@ SsoElements SolarSystemEditor::readMpcOneLineMinorPlanetElements(const QString &
...
 		objectType = "sednoid";
 
-	//Radius and albedo
-	//Assume albedo of 0.15 and calculate a radius based on the absolute magnitude
-	//as described here: http://www.physics.sfasu.edu/astro/asteroids/sizemagnitude.html
-	double albedo = 0.15; //Assumed
-	double radius = std::ceil(0.5*(1329 / std::sqrt(albedo)) * std::pow(10., -0.2 * absoluteMagnitude)); // Original formula is for diameter!
-	result.insert("albedo", albedo);
-	result.insert("radius", radius);
+	// don't save estimated albedo and radius
```
Comment (gzotti) [inline plugins/SolarSystemEditor/src/SolarSystemEditor.cpp:1198]: This comment will not be understood a week from now. Suggestion:

// don't estimate and save unknown albedo and radius here. These will be handled in the loader.

**Review by gzotti — APPROVED** [review]
Thanks for this! Apart from comments and a hoped good source for the albedo estimate, looks good to me.

File: src/core/modules/MinorPlanet.hpp (hunk 1 above)
Comment (mgarraha) [reply src/core/modules/MinorPlanet.hpp]: Updated comment. Uses ssystem_minor.ini albedo if provided.

File: src/core/modules/SolarSystem.cpp (hunk 2 above)
Comment (mgarraha) [reply src/core/modules/SolarSystem.cpp]: Might as well default to 0 here. Added citation to updateEquatorialRadius.

File: plugins/SolarSystemEditor/src/SolarSystemEditor.cpp (hunk 3 above)
Comment (mgarraha) [reply plugins/SolarSystemEditor/src/SolarSystemEditor.cpp]: Revised comment.

**Pull request merged**